sandbox/lua/decoders/bind_query_log.lua

```diff
@@ -0,0 +1,274 @@
+-- '('
+local open_paren_literal = l.P"("
+-- ')'
+local close_paren_literal = l.P")"
+-- 'query'
+local query_literal = l.P"query:"
+-- 'IN' literal string; 
+local in_literal = l.P"IN"
+-- '+' literal character; + indicates that recursion was requested.
+local plus_literal = l.P"+"
+-- '-' literal character; - indicates that recursion was not requested.
+local minus_literal = l.P"-"
+-- 'E' literal character; E indicates that extended DNS was used
+-- Source: http://ftp.isc.org/isc/bind9/cur/9.9/doc/arm/Bv9ARM.ch06.html#id2575001
+-- More about EDNS: https://en.wikipedia.org/wiki/Extension_mechanisms_for_DNS
+local e_literal = l.P"E"
+-- 'S' literal character; s indicates that the query was signed
+-- Source: http://ftp.isc.org/isc/bind9/cur/9.9/doc/arm/Bv9ARM.ch06.html#id2575001
+local s_literal = l.P"S"
+--'D' literal character; D means the client wants any DNSSEC related data
+-- Source: http://ftp.isc.org/isc/bind9/cur/9.9/doc/arm/Bv9ARM.ch06.html#id2575001
+local d_literal = l.P"D"
+--'T' literal character; if TCP was used
+-- Source: http://ftp.isc.org/isc/bind9/cur/9.9/doc/arm/Bv9ARM.ch06.html#id2575001
+local t_literal = l.P"T"
+--'C' literal character; queryer wants an answer anyway even if DNSSEC validation checks fail
+-- Source: http://ftp.isc.org/isc/bind9/cur/9.9/doc/arm/Bv9ARM.ch06.html#id2575001
+local c_literal = l.P"C"
+
+
+--[[ More complicated patterns for things that do change from line to line: --]]
+
+--The below pattern matches date/timestamps in the following format:
+-- 27-May-2015 21:06:49.246
+-- The milliseconds (the .246) are discarded by the `l.P"." * l.P(3)` at the end:
+--Source: https://github.com/mozilla-services/lua_sandbox/blob/dev/modules/date_time.lua
+local timestamp = l.Cg(date_time.build_strftime_grammar("%d-%b-%Y %H:%M:%S") / date_time.time_to_ns, "Timestamp") * l.P"." * l.P(3)
+local x4            = l.xdigit * l.xdigit * l.xdigit * l.xdigit
+
+--The below pattern matches IPv4 addresses from BIND query logs like the following:
+-- 10.0.1.70#41242
+-- The # and ephemeral port number are discarded by the `pound_literal * l.P(5)` at the end:
+local client_ip = l.Cg(l.Ct(l.Cg(ip.v4, "value") * l.Cg(l.Cc"ipv4", "representation")), "ClientIP") * pound_literal * l.P(5)
+
+--The ends of query logs have the IP address the DNS server used to respond to the query with. The LPEG capture group is just like 
+-- the client IP, but encased in ( ) and without the # literal at the front: (10.0.1.71)
+local server_responding_ip = l.P"(" * l.Cg(l.Ct(l.Cg(ip.v4, "value") * l.Cg(l.Cc"ipv4", "representation")), "ServerRespondingIP") * l.P")"
+
+--[[DNS query record types:
+Create a capture group that will match the DNS record type.
+
+The + signs mean to select A or CNAME or MX or PTR and so on.
+
+The ', "record_type"' part sets the name of the capture's entry in the table of
+matches that gets built.
+
+Source: https://en.wikipedia.org/wiki/List_of_DNS_record_types
+--]]
+dns_record_type = l.Cg(
```

Comment:
A regular capture will suffice for all RecordType and RecordClass entries e.g. `l.C"A"`